Set up GitHub Actions

wooferzfg opened the pull request:
Here's a build that finished which included a deployment step: https://github.com/wooferzfg/livesplitone/commit/b51f1b1987405c1654a2c7945eb30622e6cd8ae6/checks?check_suite_id=408981317

Here's another build that finished which used the cached directories: https://github.com/wooferzfg/livesplitone/commit/b1cf4373ece2560920945c64c0a402548047c6c2/checks?check_suite_id=409000668

Fix #174

CryZe commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,71 @@
+        mv dist/* .
+        git add :/
+        git commit -m "GitHub Pages build"
+        git push --force https://action:${{ secrets.LIVESPLIT_ONE_DEPLOY }}@github.com/LiveSplit/LiveSplitOne.git
```
Is there no prebuilt Action for pushing?

wooferzfg replied:
There is, but I had trouble getting it to work when I was setting up the build to push the `livesplit-core` docs. I'll try it again though.

CryZe commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,71 @@
+    - name: Install Node
+      uses: actions/setup-node@v1
+      with:
+        node-version: '12.x'
```
I'd prefer not to specify a version at all / always choose the latest Node.

wooferzfg replied:
The default version for `actions/setup-node` is Node 10, but I think we want to be on Node 12 (since it's the latest LTS release). I don't think this action lets you specify "latest LTS" any other way besides `12.x`.

CryZe replied:
Is there an issue tracking this there? Or should we maybe open one?

wooferzfg replied:
https://github.com/actions/setup-node/issues/26

CryZe replied:
Can you mark it with a comment `FIXME: https://github.com/actions/setup-node/issues/26` so we remember checking this.

CryZe commented on .github/workflows/install.sh:
```diff
@@ -1,23 +1,15 @@
-    rustup target install $TARGET
-
     cargo install -f wasm-gc
     cargo install -f wasm-bindgen-cli
```
I believe we may not need the -f anymore. In Rust 1.41 (which this being nightly should be 1.42) it supposedly is automatically updating on cargo install (but only if it's necessary). And since we are caching, this should reduce CI build times a lot.

CryZe replied:
Seems like this works:

First it upgrades from .57 to .58:
![https://i.imgur.com/GvA5zsD.png](https://i.imgur.com/GvA5zsD.png)

And then it refuses to upgrade any further:
![https://i.imgur.com/bSo1v1P.png](https://i.imgur.com/bSo1v1P.png)

CryZe commented:
The badge in the README should be replaced then too.

CryZe commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,71 @@
+    branches:
+      - 'master'
+    tags:
+      - '*'
```
I don't think we want it on tags.

wooferzfg commented:
Updated build with deployment: https://github.com/wooferzfg/LiveSplitOne/commit/f6097dd455226c4ab00dd35e2b50e7ba294407a8/checks?check_suite_id=409025713

Updated build with cache: https://github.com/wooferzfg/LiveSplitOne/commit/aec2538f7f327f566ba362f9b0c61b4194f0c947/checks?check_suite_id=409044777

CryZe commented on .github/workflows/install.sh:
```diff
@@ -1,25 +1,17 @@
+    cargo install wasm-bindgen-cli
 
     # This fetches latest stable release
     local tag=$(git ls-remote --tags --refs --exit-code https://github.com/japaric/cross \
```
Why are we even installing cross? That's from way back when we used emscripten. I believe we can throw this out entirely too.

CryZe commented on .github/workflows/install.sh:
```diff
@@ -1,25 +1,17 @@
-        sort=sort
-    fi
+    local target=x86_64-unknown-linux-musl
+    local sort=sort
```
This stuff up here is also for cross.

CryZe commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,76 @@
+      with:
+        branch: 'gh-pages'
+        force: true
+        github_token: ${{ secrets.LIVESPLIT_ONE_DEPLOY }}
```
The reason why I've been asking if there's not a pre-existing one was because I believe we don't need to provide our own token. GH Actions already come with their own one. So I believe we should use that one and remove this one entirely.

CryZe replied:
https://github.com/LiveSplit/livesplit-core/blob/master/.github/workflows/rust.yml#L677

wooferzfg replied:
Oh yeah, you're totally right. We only needed our own token in `livesplit-core` because we were uploading to a different repo.
